### DISCUSSION TITLE
24.XX and Re Running UBI Installer - Some Noob Questions

### DISCUSSION THREAD

**Discussion by ghost** [dangowrt/owrt-ubi-installer#182] [Q&A] [answered]
Hello all,

Looking at the [OpenWRT page here](https://openwrt.org/toh/linksys/e8450), I see that I need to "re run the UBI Installer" before upgrading to any future 24.xx release of OpenWRT on this router.  I also see that there is an experimental release of the installer on the [Releases Page here](https://github.com/dangowrt/owrt-ubi-installer/releases) so I assume I will need to come back here to this GitHub to get my router's UBI tables up to date.

Anyway, some noob questions since I'm new to this device:

1. Can (in the future when a stable 24.xx comes out) I run this installer on a router that is already running OpenWRT, or do I need to go back to stock first?  The GitHub guide only covers initially going to OpenWRT from stock.
2. If the answer is Yes in that I can do a system/UBI upgrade from OpenWRT, do I follow the same instructions for stock?  Meaning: recovery ITB then sysupgrade ITB?  Or do I just go straight to the sysupgrade ITB?
3. Lastly, how can I tell if I need to use signed or unsigned firmware for my device?  I bought this router off of eBay and to my surprise it came preloaded with OpenWRT already.  I don't know what stock firmware this was on beforehand, as in if it was above that 1.2 threshold for signed firmware outline in the guide.

Thank you in advance.

**Comment by dangowrt** [comment, marked as answer]
> 1. Can (in the future when a stable 24.xx comes out) I run this installer on a router that is already running OpenWRT, or do I need to go back to stock first?  The GitHub guide only covers initially going to OpenWRT from stock.

You don't need to return to stock. You can force-flash the new installer while already running OpenWrt 23.05.x.

> 3. Lastly, how can I tell if I need to use signed or unsigned firmware for my device?

You will need to use the unsigned version of the installer.

**Reply by ghost** [reply]
Thanks for the quick reply.  If you don't mind, can you update your readme to say that the installer can be run on OpenWRT as a normal upgrade?  It'll probably help others like me that aren't too familiar with this device and the process.

**Comment by Falcon-118** [comment]
Dan, first off - thank you for your efforts.  Without you the e8450/rt3200 would probably never see OpenWRT working.    I'd like to get some clarification on your advice to Jason.  I have a handful of e8450s as I live in a large house stretched out across a mountainside.  I don't use them as routers, just WAPs behind a pfSense box.  I also make heavy use of DSA for VLAN distribution, wired and wireless.    My first build was about this time last year but I'm not sure if it was your ver. 101 or 102 and the latest OpenWRT version at that time.  I did not follow your detailed instructions, rather a youtube video.    It seemed to work for my needs but it got back burnered until this spring when I finally decided the WRT3200ACMs were having too many WiFi glitches.    I built 5 in total, the last 4 with your ver. 1.03 and following your instructions to the letter.   About 3 weeks into use the one in my shop (first one, NOT built on 1.03) started showing wifi errors and hanging a transported VLAN ethernet connection.   After poking around, I couldn't find the cause so I rebooted it.  DEAD/KOD - NO lights.   Power cycled it, DEAD/no lights.  I had good backups so the spare built with 1.03 went in to replace it with a restore.  It's been fine ever since.

This week I opened up the dead e8450 and dug out my usb/ttl adapter and much to my surprise/relief I found the JTAG could see about 8 lines of code would run and then just halt... still, no lights.  It would never get to the eleven item boot menu.  I thought I took a snapshot of the screen, but now I can't find it.  So, using the mtk_uartboot I was able to recover it using the .fip file from the 23.05.3 files - everything seems to be intact.  I have run it through several reboots just fine.  I've done a complete reset and it sill comes back up, all lights working.  

So, back to your message to Jason.  Do you recommend leaving as is or rerunning through your setup instructions using the 1.03 boot loader again?

Thanks!
